secboot/secboot.go

```diff
@@ -143,3 +143,7 @@ type UnlockResult struct {
 	// - UnlockedWithKey
 	UnlockMethod UnlockMethod
 }
+
+// FDEHasReveal is setup by devicestate/fde to support device-specific
+// full disk encryption implementations.
+var FDEHasRevealKey = func() bool { return false }
```

Comment:
Should this panic instead to alert us immediately if the setup is broken, or is it also a default fallback when not provided for specific devices?

Reply:
We could, we need to mock more in the teststhen

Reply:
it would be just:
```
diff --git a/secboot/secboot.go b/secboot/secboot.go
index 3c2149b126..41d2f4a1ed 100644
--- a/secboot/secboot.go
+++ b/secboot/secboot.go
@@ -146,4 +146,4 @@ type UnlockResult struct {
 
 // FDEHasReveal is setup by devicestate/fde to support device-specific
 // full disk encryption implementations.
-var FDEHasRevealKey = func() bool { return false }
+var FDEHasRevealKey = func() bool { panic("FDEHasRevealKey not set"); return false }
diff --git a/secboot/secboot_tpm_test.go b/secboot/secboot_tpm_test.go
index 3f5edb0269..a32e1c7cf6 100644
--- a/secboot/secboot_tpm_test.go
+++ b/secboot/secboot_tpm_test.go
@@ -58,6 +58,9 @@ var _ = Suite(&secbootSuite{})
 func (s *secbootSuite) SetUpTest(c *C) {
        dirs.SetRootDir(c.MkDir())
        s.AddCleanup(func() { dirs.SetRootDir("/") })
+
+       secboot.FDEHasRevealKey = func() bool { return false }
+       s.AddCleanup(func() { secboot.FDEHasRevealKey = nil })
 }
 
 func (s *secbootSuite) TestCheckKeySealingSupported(c *C) {
```
Happy to add it @pedronis wdyt?

Reply:
HasFDESetupHook is also set to return false, I don't have super strong opinions, but in this case the packages work fine with those functions returning false, it's a problem of the using packages to change that if they need the hook functionality